### PR TITLE
Adding event to billing deleting of payment

### LINKED ISSUE #6841: Add an event to billing payment deletion
### Describe the problem

For the credit card processing, it would be nice to have an event fired off. T

### Describe the solution you'd like

Build event class.
Add event dispatch to the search payments where the patient payments are deleted

### PR DESCRIPTION
**Pull request by juggernautsei**
Fixes #6841

#### Short description of what this resolves:
Add class for billing event when payments are deleted

#### Changes proposed in this pull request:
Edits to search_payment.php
New class added to the Events folder.

### CODE REVIEW

File: interface/billing/search_payments.php [hunk 1]
```diff
@@ -51,7 +57,8 @@
                 sqlStatement("update form_encounter set last_level_closed=last_level_closed - 1 where pid =? and encounter=?", [$PId, $Encounter]);
             }
         }
-
+        //dispatch this payment is being deleted trigger refund process
+        $eventDispatcher->dispatch(new DeletePayment(add_escape_custom($DeletePaymentId)), DeletePayment::ACTION_DELETE_PAYMENT, 10);
```
Comment (adunsulag) [inline interface/billing/search_payments.php:61]: I wouldn't escape the payment id here.  Downstream consumers will then have to remove the escaping it if they use it for something different than database insertion.

**Review by adunsulag — CHANGES_REQUESTED** [review]

File: interface/billing/search_payments.php (hunk 1 above)
Comment (juggernautsei) [reply interface/billing/search_payments.php:61]: I understand what you are saying, and in my use case, that'd be great. I only use the ID to retrieve information once I determine that the credit card processor makes the entry in the database.

**Review by adunsulag — APPROVED** [review]
This looks good to me.

**Comment by bradymiller** [comment]
dance-bear

**Pull request merged**